<?xml version="1.0" ?>

<launch>
    <arg name="task_file" default="$(find cbf_controllers)/config/task.info"/>
    <arg name="reference_file" default="$(find cbf_controllers)/config/reference.info"/>
    <arg name="urdf_file" default="$(find legged_unitree_description)/urdf/a1.urdf"/>
    <arg name="gait_file" default="$(find legged_controllers)/config/gait.info"/>

    <!-- make the files into global parameters -->
    <param name="task_file" value="$(arg task_file)"/>
    <param name="reference_file" value="$(arg reference_file)"/>
    <param name="urdf_file" value="$(arg urdf_file)"/>
    <param name="gaitCommandFile" value="$(arg gait_file)"/>

    <rosparam file="$(find cbf_controllers)/config/default.yaml" command="load"/>

    <node pkg="ocs2_legged_robot_ros" type="legged_robot_gait_command" name="legged_robot_gait_command"
          output="screen"/>

    <node pkg="rostopic" type="rostopic" name="robot_visualiser"
          args="pub /robot_marker visualization_msgs/Marker
  '{header: {seq: 0, stamp: {secs: 0, nsecs: 0}, frame_id: 'base'},
    ns: '', id: 0, type: 4, action: 0,
    pose: {position: {x: 0.0, y: 0.0, z: 0.0}, orientation: {x: 0.0, y: 0.0, z: 0.0, w: 1.0}},
    scale: {x: 0.02, y: 0.0, z: 0.5},
    color: {r: 0.7, g: 0., b: 0., a: 1.0},
    lifetime: {secs: 0, nsecs: 0},
    points: [{x: 0.25, y: 0.16, z: 0.0}, {x: 0.25, y: -0.16, z: 0.0}, {x: -0.35, y: -0.16, z: 0.0}, {x: -0.35, y: 0.16, z: 0.0}, {x: 0.25, y: 0.16, z: 0.0} ],
    frame_locked: true }' -r 2" output="screen"/>

    <node pkg="cbf_controllers" type="visualiser" name="visualiser"/>
</launch>
